### DISCUSSION TITLE
After reloading a model the representative docs are not available anymore

### DISCUSSION THREAD

**Discussion by v1alina** [MaartenGr/BERTopic#2046] [Q&A] [answered]
Hey Maarten,

I hope you can help me with an issue I have when reloading my topic model.

I saved my model using Pytorch with both ctf-idf and the embedding model saved as well. 
Loading it works just fine but when I run `topic_model.get_topic_info()` it works fine but the Representative_Docs column only has NaN values. Is that a bug and more importantly is there are way I can get the top three docs for each topic assigned to the column again? 

I'm on version 0.16.2

**Comment by MaartenGr** [comment, marked as answer]
I believe that the representative documents are currently not saved for privacy reasons as they might contain sensitive data. Instead, I would advise saving them separately and then assigning them to `topic_model.representative_docs_` when loading the model. Other than that, you can re-calculate the representative documents with `_extract_representative_docs` but do note that it is a private function which might change in future versions.

**Reply by 6801318d8d** [reply]
Can you provide more guidance on how to use `_extract_representative_docs()`? It requires 3 positional parameters: 'c_tf_idf', 'documents', and 'topics'

**Reply by MaartenGr** [reply]
The topics and c_tf_idf are the attributes of the topic model, so you can access those directly. The documents need to be processed according to the lines here:

https://github.com/MaartenGr/BERTopic/blob/04c52a83c0865cb25db785e4403cc9f789e2ddd5/bertopic/_bertopic.py#L448

Just make sure you also add the topic attributes of the model to the "Topic" column.

**Reply by 6801318d8d** [reply]
Can we update the Serialization docs to specify how a full "safetensors" save should be performed?

**Reply by 6801318d8d** [reply]
> The topics and c_tf_idf are the attributes of the topic model, so you can access those directly.

Maybe I succeeded, but I'm not sure.

Is this right?

```
documents = pd.DataFrame({"Document": indf["text"].tolist(), "ID": range(len(indf["text"].tolist())), "Topic": topic_model.topics_, "Image": None})
res = topic_model._extract_representative_docs(topic_model.c_tf_idf_, documents, topics = topic_model.topic_representations_)
topic_model.representative_docs_ = res[0]
```

**Reply by MaartenGr** [reply]
> Is this right?

Yes, I believe it is.

> Can we update the Serialization docs to specify how a full "safetensors" save should be performed?

Not in its current form. This is a private function that is not meant to be used as I cannot promise a stable API (which you should not for private functions). Additional functionality might be possible at some point to recalculate the representative documents, but BERTopic already has way more functionality than I had expected when I first created it, so it's becoming increasingly difficult to maintain with more features. 

If there is a simple/elegant solution for this, I'm all ears!
